LyricWiki php fixes

SeanColombo opened the pull request:
Fixes LyricWiki code issues to kill warnings/errors and deprecations related to upgrading to PHP 5.6.

LYR-231, LYR-232, LYR-233

Grunny committed:
(CE-1391) Initial commit of DMCA Request extension

macbre commented on extensions/3rdparty/LyricWiki/server.php:
```diff
@@ -2181,10 +2181,10 @@ function lw_getTitle($artist, $song='', $applyUnicode=true, $allowAllCaps=true){
+	$title = preg_replace_callback('/([-\("\.\/:_])([a-z])/', function($m){ return $m[1].strtoupper($m[2]); }, $title);
+	$title = preg_replace_callback('/\b(O)[\']([a-z])/i', function($m){ return $m[1].strtoupper("'".$m[2]); }, $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.  Does it to "O'Riley" but not "I'm" or "Don't"
+	$title = preg_replace_callback('/( \()[\']([a-z])/i', function($m){ return $m[1].strtoupper("'".$m[2]); }, $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.
+	$title = preg_replace_callback('/ [\']([a-z])/i', function($m){ return " ".strtoupper("'".$m[1]); }, $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.
```
This code should be unit tested :)

macbre commented on extensions/3rdparty/LyricWiki/server.php:
```diff
@@ -2141,7 +2141,7 @@ function lw_fmtAlbum($album,$year){
+	$retVal = preg_replace_callback('/([-\("\.])([a-z])/', function($m){ return $m[1].strtoupper($m[2]); }, $retVal);
 	$retVal = str_replace(" ", "_", $retVal);
 	return $retVal;
 } // end lw_fmtSong()
```
I think it's a good moment to add unit tests here.

macbre replied:
@SeanColombo - ping :)

macbre replied:
Ping

Commit:
ADEN-1748 EXPERIMENT create ad provider OpenX (URL forceopenx=1)

Commit:
ADEN-1748 Include INCONTENT_1* ads in OpenX

lukaszkonieczny committed:
CE-1419: Fix fatal error in SMW

Commit:
ADEN-1748 Updated unit tests

Commit:
SOC-628 Adding use of WikiaDataAccess

Commit:
ADEN-1947 fixed URL param so it's stronger ;)

Commit:
ADEN-1947 fixed PHP unit tests

Commit:
SOC-605 Merge part #1

lukaszkonieczny committed:
CE-1639: Changed way to support links

lukaszkonieczny committed:
CE-1639: Removed spaces

lukaszkonieczny committed:
CE-1607: Get next item to fix from insight list

lukaszkonieczny committed:
CE-1607: Encode string

Commit:
SOC-674 log which user properties are being used

Commit:
SOC-647 use correct property number count

Commit:
SOC-674 make logging key one word

Commit:
ADEN-1947 hooks work better when registred

bognix committed:
CONCF-477 initial work for backgrounds on breakpoints

bkoval committed:
Use correct argument when enqueuing and dont load msgs in js

bkoval committed:
Updating correct message

macbre committed:
eventsLocalUsersCleanup: initial commit

bognix committed:
CONCF-477 move common values to vars and clean up

Commit:
PLATFORM-1160 Uses $status->value for comparisons...

... and messaging rather than $status.

dianafa committed:
enable social icons worldwide

dianafa committed:
add pageshare setup

dianafa committed:
ad variable in defaultsettings.php

macbre committed:
eventsCleanup: cleanup the events table as well

Commit:
SOC-675 Fix reversed logic

dianafa committed:
show line only for touch screens & tests

dianafa committed:
remove witespace

hakubo committed:
spaces in function call

hakubo committed:
spaces in function definition

Commit:
CONCF-483 update whitespaces

Commit:
These classes are no longer relevant and were supposed to be removed a long time ago

Commit:
CONCF-477 | scss lint fixes

lukaszkonieczny committed:
CE-1641: Disable create new wiki popup in special Insights

lukaszkonieczny committed:
CE-1641: Change if statement

lukaszkonieczny committed:
CE-1641: Add more comments

Commit:
SOC-605 Move translations to the right place

Commit:
SOC-396 Limit large queries for wall thread replies

Commit:
SOC-396 Add changes lost by my editor

Commit:
SOC-396 Change param type

Commit:
ADEN-1971 Non-critical tweaks to Paid Asset Drops

Commit:
Use a different key while logging to ELK, 'status' will not work

macbre committed:
WikiService::getTopEditors - improve DB profiling

macbre committed:
WikiService: use $wgSpecialsDB instead of a hardcoded name

bognix committed:
CONCF-477 fix scss-lint warnings when possible

Commit:
CONCF-146 update translations

dianafa committed:
change true/false string to 0/1

macbre committed:
TransactionClassifierTest: add test cases for Nirvana API classifiers

macbre committed:
TransactionClassifier: monitor Nirvana controllers used by Mobile Apps

dianafa committed:
remove npm-log file

dianafa committed:
merge with dev

dianafa committed:
Revert "merge with dev"

This reverts commit 0c602750222c2ec47273aa9da8ef99c090881817.

dianafa committed:
add removed comment

bognix committed:
CONCF-483 fix caching for touchscree/non-touchscreen versions

Commit:
SOC-396 Feedback from code review

Commit:
disable index/key use (temporarily) MAIN-4386

Commit:
fix loop output in maintenance script

adamkarminski committed:
CE-1634 Attach the rail module in a hook

Commit:
SOC-683 Use unset() directly on global var

kamilkoterba committed:
CE-1623 Adding clicktracking for items on special insights page

Commit:
ADEN-1975 Prioritize Xaxis segment in Krux segment for passing to DFP

lukaszkonieczny committed:
CE-1641: Comments added

kamilkoterba committed:
CE-1623 Sanitizing output

kamilkoterba committed:
CE-1623 Removing variable that isn't reused anywhere else

Commit:
ADEN-1977 Drop the Paid Asset to main page left column

adamkarminski committed:
CE-1694 Load right rail module with the extension and fix messages

adamkarminski committed:
CE-1645 Use getMessagesKeys in subpageList

kamilkoterba committed:
CE-1623 Adding tracking for special insights index page

kamilkoterba committed:
CE-1622 Adding onShow callback to BannerNotification

kamilkoterba committed:
CE-1622 Adding click tracking on next-page and back-to-list buttons on Insights loop notifications

bognix committed:
CONCF-477 add missing class to body

kamilkoterba committed:
CE-1622 Add tracking for close button on Insights notification

Commit:
Put CuratedContent into Mercury API getArticle

vforge committed:
CONCF-444 | Cleanup GlobalNavigation code

kamilkoterba committed:
CE-1614 Added tracking of success banner impression

Commit:
SERVICES-292 CuratedContent::getList should return allcategories if there is no data in wgCuratedContent

lukaszkonieczny committed:
CE-1644: Display proper notification in VE

Commit:
DAT-2748 Handle null arguments properly; Add tests

adamkarminski committed:
CE-1650 Complete documentation for the Insights feature

adamkarminski committed:
CE-1648 Reduce small icons size to 21px

vforge committed:
CONCF-475 | Fix H2 in PageHeader

kamilkoterba committed:
CE-1614 Distinguish tracking for regular fixed status and alldone status

Commit:
CONCF-475 | [minor] removed an empty line

vforge committed:
No not wrap user menu items

adamkarminski committed:
CE-1557 Throw exception if the StatsDB is not enabled

Commit:
CONCF-474 | [minor] properties order, empty lines

Commit:
Portable Infoboxes - initial release behind WF flag

lukaszkonieczny committed:
CE-1644: Removed checking if item is fixed from hook

Grunny committed:
(CE-1534) Add undeleted images back to the image review queue

When a deleted file is undeleted, it should be readded to the image review
queue. This adds the image data with the undeletion and fixes a bug in the
undeletion event due to incorrectly checking for an instance of article.

bognix committed:
CONCF-487 pass sections and items from curated content to Mercury

Commit:
remove code from previos infobox service approach

bognix committed:
Remove memcache from social icons

Commit:
SOC-641 Use a different subject line for anonymous user emails

Commit:
SOC-605 Downloaded new translations from LOC-32

Commit:
SOC-641 Merge part #1

Commit:
ADEN-1980 Disaster recovery for Paid Asset Drop

Commit:
SOC-396 Use alternative array syntax

Commit:
SOC-396 Rename constant

Commit:
SOC-396 Reset data when cache is cleared (and DRY)

Commit:
ADEN-1980 Unit tests updated

Commit:
Fix unit test warning

Commit:
fix ProfileOut call

lukaszkonieczny committed:
CE-1644: constants added

lukaszkonieczny committed:
CE-1644: Removed empty line

adamkarminski committed:
CE-1651 Add a zero state message

bognix committed:
CONCF add caching to PageShareController response

vforge committed:
Chain method call

bognix committed:
CONCF-496 add encoding for title and url

macbre committed:
ArticlesUsingMediaQuery: use a standard format of memcache keys

Commit:
Track when user clicks the Save button on Special:CuratedContent

kamilkoterba committed:
CE-1623 Run init functions on DOM ready

lukaszkonieczny committed:
CE-1644: Added spaces

kamilkoterba committed:
CE-1623 Track on mousedown and keydown instead of click

kamilkoterba committed:
CE-1623 move next button focus inside onShow event

kamilkoterba committed:
CE-1623 Init bn var right away where it's declared

macbre committed:
MemcachedStats: improve keys normalization

* treat : as a key parts separator
* normalize using prefixes

bognix committed:
CONCF-487 add caching for curatedContent and resolve article id to article url

adamkarminski committed:
CE-1661 Add Insights to Toolbar and Admin Dashboard

macbre committed:
MemcachedStats: treat _ as a separator

adamkarminski committed:
CE-1661 Change SpecialPage group to wikia

kamilkoterba committed:
CE-1623 Use on function instead binding many times within each

kamilkoterba committed:
CE-1626 Adding proper link to insights help page

adamkarminski committed:
CE-1662 Update messages

macbre committed:
MemcachedStats: more keys to normalize

adamkarminski committed:
CE-1671 Fix XSS vulnerability

macbre committed:
MemcachedStatsTest: add a new line at the end

dianafa committed:
skip wfFindFile function and add throbber

adamkarminski committed:
CE-1670 Add See more insights link

dianafa committed:
working throbber

dianafa committed:
remove comment

lukaszkonieczny committed:
CE-1623: Set tracking for loop notifications

dianafa committed:
add throbber to hub video upload

lukaszkonieczny committed:
CE-1623: Combine tracking for index and list Insights pages

Commit:
SERVICES-292 CuratedContent should return allcategories in same format as from wgCuratedContent

Commit:
SOC-579 Fix strict warning for extra function param

timmyquivy committed:
MAIN-4392

Adding ability for staff and helpers to add local threadmoderators.

Commit:
SOC-579 Variable is initialized in constructor

Commit:
HG-600: Add information about section indexes to WikiaMobile and ArticleAsJson output

Given the fact that WikiaMobile and ArticleAsJson are to be retired - and latter to be replaced by Article Service - this seems like a right solution

This change is also unifying output for ArticleAsJson so it does not depend on user state (logged in or out).

kamilkoterba committed:
CE-1623 No need for double escaping: Changing escaped to plain as messages are escaped in mustache template by {}

bognix committed:
CONCF-487 change API to return sections and items in separate requests

lukaszkonieczny committed:
CE-special-insights: Do not allow admins modify redirect url

macbre committed:
FluentSQL: assert that we get a result wrapper

Commit:
PLATFORM-1151 refresh the db state after Helios call

In the scope of a DB connection, once the first read occurs, MediaWiki
will operate on a frozen state of the DB and will not see changes made
by external services. I added an explicit commit that will force
refreshing the state after Helios call.

macbre committed:
BatchRefreshLinksForTemplate: better structured logs from batchEnqueue

macbre committed:
BaseTask: log when the task is queued

bognix committed:
CONCF-487 separate APIs for sections and items

macbre committed:
LinksUpdate::queueRefreshTasks - remove legacy jobs no-op

kamilkoterba committed:
CE-1626 Fix link to help page

macbre committed:
LinksUpdate: log BatchRefreshLinksForTemplate tasks

lukaszkonieczny committed:
CE-1657: Add flow to all edit entry points on page

Commit:
CE-1699 new messages for translation

Commit:
SOC-701 Results count needs to be an integer

Commit:
Improve the qqq for a Comment email i18n key

Grunny committed:
(CE-1391) Finalising initial version of request form

Grunny committed:
(CE-1679) Remove Phalanx block from the rename process

When the local rename part of the rename process fails at any point
the user is regularly left blocked until we can clean up the rename.
This is unnecessary as there is no reason to stop the user making use
of the new name after the global rename has happened but before all
the edits and so forth have been reattributed.

This removes the Phalanx block as when it is put in place, the main
rename that updates the user table and creates a fake user at the old
name has already occurred.

adamkarminski committed:
CE-1683 Page views display implementation

macbre committed:
wgUpdateRowsPerJob: throttle number of pages to update per job / task

macbre committed:
Log tasks execution + consistent logging in BaseTask

Commit:
SOC-637 | GlobalWatchlistTask:addWatchers query fix

Commit:
Revert "SOC-674 log which user properties are being used"

Commit:
SOC-682 | UserSignup fix page after POST request

adamkarminski committed:
CE-1682 Simplify retrieving last timeIds

Commit:
SOC-631 | Email - fix cursor on link in email

Commit:
DAT-2752 Stop Hero Module progression;

* Do not enable on new wikis
* Allow only for opt-out in Wikia Labs
l

Commit:
SOC-654 | Email - button position

Commit:
SOC-631 | Email - fix cursor on link in email, rename CSS class

macbre committed:
DatabaseMysqli::mysqlRealEscapeString - assert a proper argument

lukaszkonieczny committed:
CE-1715: Fix Fatal Error on Insights

macbre committed:
robots.txt: set Crawl-delay fro msnbot

lukaszkonieczny committed:
CE-1657: Update doc blocks

lukaszkonieczny committed:
CE-1657: Remove typo

Commit:
MAIN-4429 Updated translations

Grunny committed:
(CE-1714) Fix removal of fixed items from Insights

Because of the way our DB transactions are handled, the removal of
fixed items from the Insights list was failing due to the changes
not being committed. This fixes it by both explicitly committing the
changes and switching the request to a POST (which would auto-commit
at the end of the request anyway) as it performs a write action.

Also fixes a small bug where the deletion did not include the namespace
of the article.

Commit:
SOC-747 Displaying the correct comment now.  Also hopefully fixed race condition

Commit:
ADEN-1994 Adding INVISIBLE_SKIN to the Turtle mix

Commit:
SOC-747

* Remove unneeded member variable
* Updated comment that referred to old code
* Make sure not to query solr in dev

Commit:
SOC-747 Remove unecessary constant & don't query Solr in Dev.

Commit:
MAIN-4436 Don't check providers here

Wikia vidoes (both local and in the library) come through here and will fail this check because they are not listed as an external provider

siebrand committed:
Localisation updates from https://translatewiki.net.

macbre committed:
LoadBalancer: log 1% of connections

Commit:
ADEN-1994 Update the INVISIBLE_SKIN size list for Turtle

Nef: Please make skin size 1000x1000,1x1

lukaszkonieczny committed:
CE-1682: Add method to get insights memcache key

adamkarminski committed:
CE-1683 Improve the page views gathering mechanism

lukaszkonieczny committed:
CE-1712: Purge Insights cache

lukaszkonieczny committed:
CE-1712: Change method names

adamkarminski committed:
CE-1682 Make articles page views method more universal

adamkarminski committed:
CE-1682 Only fetch page views if any article exists

adamkarminski committed:
CE-1682 getContent should just return [] if no articles were fetched

Commit:
fix typo in class name

Commit:
add additional renderService tests

adamkarminski committed:
CE-1682 Add documentation

macbre committed:
Revision: cache blobs in memcache

Set $wgRevisionCacheExpiry to an hour

lukaszkonieczny committed:
CE-1586: Fix CNW headline message

Commit:
Revert "DatabaseMysqli::mysqlRealEscapeString - assert a proper argument"

bognix committed:
CONCF change track method from both to analytics

Commit:
SOC-751 Cache results of getBanInfo

Also cleanup file a bit

bognix committed:
CONCF-487 add missing comment and remove redundant coma

bognix committed:
CONCF-487 remove redundant function and ws fixes

Commit:
Differentiate between successful and failed save

Commit:
SOC-747 Fix some tests that expected the Solr search to be called

Commit:
Use 100% sample rate for GA

Commit:
SOC-751 Added some doc comments

Commit:
SOC-751 Downloaded new messages

Commit:
Revert "SOC-751 Downloaded new messages"

This reverts commit 7860856b74773ad07689ae03f49d512238621d41.

adamkarminski committed:
CE-1709 Implement final designs of a list with page views

macbre committed:
wgRevisionCacheExpiry: cache revision blobs for a day

lukaszkonieczny committed:
CE-1663: Add translations for mobile and desktop preview

lukaszkonieczny committed:
CE-1663: Update Edit Preview messages

adamkarminski committed:
CE-1709 Title ellipsis overflow

lukaszkonieczny committed:
CE-1660: Add pagination

lukaszkonieczny committed:
CE-1660: Change limit values

lukaszkonieczny committed:
CE-1660: Add spaces

Commit:
SOC-757 | Use slave for fetching notifications count

Commit:
Update WallNotifications.class.php

lukaszkonieczny committed:
CE-1660: Change methods class

lukaszkonieczny committed:
CE-1660: Change limit values

macbre committed:
DatabaseMysqli::mysqlRealEscapeString - log strange arguments that can't be escaped

macbre committed:
specials-schema.sql: new schema for events_local_users table

Commit:
Use AMD module and DRY

Commit:
SOC-751 Remove another use of DB_MASTER that is not needed.

Commit:
Clean up and return article_local_url for the featured section too

lukaszkonieczny committed:
CE-special-insights: Fix cache update after item is fixed

macbre committed:
robots.txt: increase the msnbot delay to 30 seconds

macbre committed:
Wall: cache AJAX responses

* cache $wn->getCounts results for 5 minutes in memcache
* cache getUpdateCounts and getUpdateWiki in the browser cache for 5 minutes
* use GET HTTP method (instead of POST)

macbre committed:
WallNotificationsExternalController: use a local cache

Commit:
ADEN-2003 Allow Evolve to hop in INVISIBLE_SKIN

lukaszkonieczny committed:
CE-insights-tracking-fix: Change tracking method

Commit:
Separate featuredContent from curatedContent

Commit:
SOC-773 | Fix SQL injection for Wall

Commit:
SOC-773 | Update whitespaces

fraszczakszymon committed:
Removed ext.wikia.adEngine.adConfigMobile definition

hakubo committed:
Wait for OO to be ready before creating a Player

Commit:
PLATFORM-1199 CloseWiki cron: remove obsolete check

We have migrated from NFS to DFS and /images is not mounted anymore
which makes is_dir check pointless and prevents the actual logic
from being executed.

macbre committed:
wgRevisionCacheExpiry: set to a month

lukaszkonieczny committed:
CE-835: Fix breaking tasks during CNW process

vforge committed:
Better TODO comment

vforge committed:
Add wgEnableMainPageDataMercuryApi

bognix committed:
Set response format to JSON

bognix committed:
Get items from curated content items response

lukaszkonieczny committed:
CE-1533: Fix Ruby formatting

fraszczakszymon committed:
Added 970x365 size to leaderboards

lukaszkonieczny committed:
CE-1736: Fix tracking impression on success banner

Commit:
PLATFORM-1032 Helios: change handling of non-200 HTTP responses

Prior this change any non-200 response from the service was
considered an error and an exception was thrown.

As of now, non-200 responses are considered legit and no exception
is thrown.

The Client class has a new property - status - and the status object
is stored there. A getter method getStatus() has been added.

Handling specific responses in specific cases has to be implemented
outside the Client class.

Commit:
do not initialize status

macbre committed:
Remove entries for wikis closed before Jun 2013

macbre committed:
EventsCleanup: clean dataware.pages table

See PLATFORM-1204

Commit:
added Insights translations

Commit:
use Sanitizer form email address validation in assertValidFromAddress

SeanColombo commented:
@macbre added unit tests :) ...interestingly, the jenkins tests are failing on a different test related to image-cropping.

I assume that's because a URL or something has changed in "dev" since I branched this. Rebasing (this is a serious rebase!)

macbre commented on extensions/3rdparty/LyricWiki/tests/ServerTest.php:
```diff
@@ -67,4 +67,30 @@ public function test_lw_getSearchResults() {
+		$this->assertEquals(lw_fmtSong("guitar%20man"), "Guitar_Man", "Testing URL decoding");
+		$this->assertEquals(lw_fmtSong("guitar (man)"), "Guitar_(Man)", "Testing various capitalization needs.");
+	}
+}
```
Thanks!

macbre commented:
LGTM

SeanColombo commented:
@macbre well, the rebasing didn't seem to help the unit-test issues. Can you still merge it easily?

macbre commented:
Please cherry pick your commits to the new branch.

SeanColombo commented:
@macbre Do you know of a way to find out what they were?  The way git does rebasing, the log is now completely garbled with other commits & I don't see a way to filter them out.  I can hard-reset back to a state before the rebase, but I'm not aware of anywhere I can actually see a list of my commits.

macbre commented:
Did just simply perform `git merge dev` on your branch?
